csharp/src/Microsoft.ML.OnnxRuntime/Training/CheckpointState.shared.cs

```diff
@@ -37,6 +37,24 @@ public CheckpointState(string checkpointPath)
             }
         }
 
+        /// <summary>
+        /// Creates CheckpointState by loading from the handle
+        /// <param name="checkpointPath"> absolute path to checkpoint file.</param>
+        /// </summary>
+        public CheckpointState(IntPtr checkpointHandle)
+            : base(checkpointHandle, true)
```

Comment:
when do we need this api?

Reply:
We needed it when `GetState` constructed a new CheckpointState object. We no longer need this.